api/api.go

```diff
@@ -76,225 +78,286 @@ func (api *ImportAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 func (api *ImportAPI) addJob(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
 	job, err := models.CreateJob(r.Body)
 	defer r.Body.Close()
 	if err != nil {
-		log.Error(err, nil)
+		log.ErrorCtx(ctx, errors.WithMessage(err, "api endpoint addJob error - Bad client request received"), nil)
 		http.Error(w, "Bad client request received", http.StatusBadRequest)
 		return
 	}
 
 	logData := log.Data{"recipeID": job.RecipeID}
 	auditParams := common.Params{"recipeID": job.RecipeID}
-	if err := api.auditor.Record(r.Context(), addJobAction, actionAttempted, auditParams); err != nil {
-		handleAuditError(w, err, logData)
+	if auditError := api.auditor.Record(ctx, addJobAction, actionAttempted, auditParams); auditError != nil {
+		handleAuditError(ctx, w, addJobAction, actionAttempted, auditError, logData)
 		return
 	}
 
-	createdJob, err := api.jobService.CreateJob(r.Context(), job)
-	if err != nil {
-		if err := api.auditor.Record(r.Context(), addJobAction, actionUnsuccessful, auditParams); err != nil {
-			handleAuditError(w, err, logData)
-			return
+	var createdJob *models.Job
+	b, err := func() (bytes []byte, err error) {
+		createdJob, err = api.jobService.CreateJob(ctx, job)
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "addJob endpoint: error creating job resource"), logData)
+			return nil, err
 		}
-		setErrorCode(w, err)
-		return
-	}
 
-	logData["job"] = job
-	auditParams["createdJobID"] = createdJob.ID
+		logData["job"] = createdJob
+		auditParams["createdJobID"] = createdJob.ID
+
+		bytes, err = json.Marshal(createdJob)
```

Comment:
I find the inline function bizarre and a little difficult to read although I can see why you have done it. Please don't call the variable name `bytes` though as it could potentially conflict with the package namespace equivalent. (I know it doesn't here, but it could do)

Reply:
Just to be clear it is recommended that you do not "function inline" if you are going to call other functions inside the inlined function. https://github.com/golang/go/wiki/CompilerOptimizations#function-inlining

Reply:
I agree it doesn't look the cleanest but it does save passing variables into a separate function and it is specific to this method.........I would possibly think of moving it out if i could come up with a name for the function that hasn't already been used and describes what it is doing.

Re. bytes, I will change this.